src/Ocelot/DependencyInjection/ConfigurationBuilderExtensions.cs

```diff
@@ -97,16 +97,16 @@ public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder
         private static IConfigurationBuilder ApplyMergeOcelotJsonOption(IConfigurationBuilder builder, MergeOcelotJson mergeTo, string json,
             string primaryConfigFile, bool? optional, bool? reloadOnChange)
         {
-            return mergeTo == MergeOcelotJson.ToMemory ? 
-                builder.AddJsonStream(new MemoryStream(Encoding.UTF8.GetBytes(json))) : 
+            return mergeTo == MergeOcelotJson.ToMemory ?
+                builder.AddJsonStream(new MemoryStream(Encoding.UTF8.GetBytes(json))) :
                 AddOcelotJsonFile(builder, json, primaryConfigFile, optional, reloadOnChange);
         }
 
         private static string GetMergedOcelotJson(string folder, IWebHostEnvironment env,
             FileConfiguration fileConfiguration = null, string primaryFile = null, string globalFile = null, string environmentFile = null)
         {
             var envName = string.IsNullOrEmpty(env?.EnvironmentName) ? "Development" : env.EnvironmentName;
-            environmentFile ??= string.Format(EnvironmentConfigFile, envName);
+            environmentFile ??= Path.Join(folder, string.Format(EnvironmentConfigFile, envName));
```

Comment:
I’m really wondering… Typically, `FileInfo` objects represent individual files and contain properties like the file name, size, and creation date. However, **don't they inherently include the folder information?**
We enumerate `FileInfo` objects on line 112.

Reply:
Yes, the `FileInfo` object contains the full path for the file. That's actually the problem.

The pattern used to check if one of the files matches one of these special names is:
```cs
if (file.Name.Equals(globalFileInfo.Name, StringComparison.OrdinalIgnoreCase) &&
    file.FullName.Equals(globalFileInfo.FullName, StringComparison.OrdinalIgnoreCase))
```

But without these changes `FullName` will never match when `globalFile` is using the default filename, because it lacks the user provided folder.

So, `file.FullName` will be something like "C:/PathToOcelotConfig/**UserValueForFolder**/ocelot.global.json" while `globalFileInfo.FullName` will be something like "C:/PathToOcelotConfig/ocelot.global.json".

That's why this is a simple fix, just prefixing the `folder` path to the default name.

This was all discussed in the original issue. You can also remove my changes and run the included test case.

Reply:
Thank you for explanation!
The problem is string concatenation, not working with `FileInfo`.
It was better to use `FileInfo` argument for the method, but it is a bit late, and we have to have a deal with `string`.

Reply:
I'm not sure I would go that far, but either way making it explicit about what path is expected for arguments like `globalConfigFile` would do.

Ok to resolve?